Make header reading compliant with RFC7230, section 3.2.2

Croug opened the pull request:
RFC7230, section 3.2.2 specifies that headers can be joined by comma separating the values so a header like this `Authorization: Bearer <JWT>, Basic <B64EncodedCreds>` is valid

pep8speaks commented:
Hello @Croug! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`flask_jwt_extended/view_decorators.py`](https://github.com/vimalloc/flask-jwt-extended/blob/ab978ee9911d72dc3bc6bd07b0081a30a5d4e6d2/flask_jwt_extended/view_decorators.py):

> [Line 189:31](https://github.com/vimalloc/flask-jwt-extended/blob/ab978ee9911d72dc3bc6bd07b0081a30a5d4e6d2/flask_jwt_extended/view_decorators.py#L189): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

vimalloc commented:
I had no idea that was valid, thanks for the pull request! If you don't mind, could you throw a  quick unit test around this this? Then I will get it merged and a new release cut ASAP.

Thanks!

Croug commented:
Yeah no problem, I will get them set up tonight.

Croug commented:
I've added the unit tests

Croug commented:
@vimalloc

vimalloc commented on flask_jwt_extended/config.py:
```diff
@@ -87,7 +87,7 @@ def header_name(self):
     @property
     def header_type(self):
-        return current_app.config['JWT_HEADER_TYPE']
+        return current_app.config.get('JWT_HEADER_TYPE')
```
Did this need to be changed? I think the `setdefault` in jwt_manager.py should guarantee that this is always present

Croug replied:
oh that was an accident, I didn't mean to change that, I had changed it to get to test for a bug, but then meant to change it back.

vimalloc commented on flask_jwt_extended/view_decorators.py:
```diff
@@ -170,25 +171,36 @@ def _decode_jwt_from_headers():
+    # <HeaderName>: <field> <value>, <field> <value>, etc...
+    if header_type:
+        field_values = split(r',\s*', auth_header)
+        jwt_header = [s for s in field_values if s.startswith(header_type)]
```
I think this could break if we had two similiar headers in here, ie `Bearer` and `Bearer2`. Although that isn't likely to happen, I wonder if we could do this check via something like `s.split[0] == header_type` instead?

Croug replied:
Yeah I wasn't thinking about that, or maybe just `s.startswith(header_type+' ')`

vimalloc commented:
Sorry for the delay in getting back to you, I was not around this weekend. I left a couple comments, if you can update those I think this looks great and I could get a new release cut.

Thanks!

Croug commented:
Alright, finally managed to fix all the pep8 issues, my was exhibiting some weird behavior that kept creating new issues, but it should all be good now, I also fixed the ambiguity call when multiple header fields have the same value and reverted the config change

vimalloc commented:
Thanks! I'll get a new release cut stat.

vimalloc commented:
Released as version 3.23.0. Thanks for contributing! :+1:
